src/messaging/ExchangeContext.h

```diff
@@ -163,7 +163,7 @@ class DLL_EXPORT ExchangeContext : public ReliableMessageContext,
         return std::move(sessionHandle.Value());
     }
 
-    bool HasSessionHandle() const { return mSession; }
+    bool HasSessionHandle() const { return (bool)mSession; }
```

Comment:
Why this cast?  If a cast _is_ needed, it should be a C++-style cast.  But it should not be needed.

Reply:
That's my bad. mSession has an `operator_bool` which looks really weird when mSession is not a bool but you return a bool. I would rather that there be a method on the object, but we can just remove `(bool)` here for now.

Reply:
done